### DISCUSSION TITLE
Using RKraft broker with GlobalEmbeddedKafkaTestExecutionListener

### DISCUSSION THREAD

**Discussion by david-szalai-otpotthon** [spring-projects/spring-kafka#3241] [Q&A] [answered]
Hey, we'd like to use the RKraft broker for our integration test with `GlobalEmbeddedKafkaTestExecutionListener`, but as the ports set with `spring.kafka.embedded.ports` are ignored, and a random port is used, we cannot define a bootstrap address for the test consumers or test appconfigs.

Is it possible to access the broker address as exposed by `embeddedKafkaBroker.getBrokersAsString()` or somehow connect to a RKraft kafka broker started by a `GlobalEmbeddedKafkaTestExecutionListener` within a `@SpringBootTest`?

**Comment by artembilan** [comment, marked as answer]
Spring Boot auto-configuration for Apache Kafka relies on the `KafkaProperties` file.
The `EmbeddedKafkaBroker` implementation expose respective Spring Boot property on its start:
```
private String brokerListProperty = "spring.kafka.bootstrap-servers";
...
if (this.brokerListProperty != null) {
	System.setProperty(this.brokerListProperty, getBrokersAsString());
}
```
So, everything you talk about has to work out-of-the-box. But yeah, only against random ports populated by the `KafkaClusterTestKit`.
That's the limitation we have to deal with for now.

**Reply by david-szalai-otpotthon** [reply]
Thanks, we set up a test consumer manually, the consumers in appcontext did work it seems out of the box, and using `System.getProperty("spring.kafka.bootstrap-servers")` solved the test consumer problem.
